src/Cake.Common/Xml/XmlPokeAliases.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+using Cake.Core;
+using Cake.Core.Annotations;
+using Cake.Core.IO;
+
+namespace Cake.Common.Xml
+{
+    /// <summary>
+    /// Contains functionality related to XML XSL transformation.
+    /// </summary>
+    [CakeAliasCategory("XML")]
+    public static class XmlPokeAliases
+    {
+        /// <summary>
+        /// Set the value of, or remove, target nodes.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="filePath">The target file.</param>
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="value">The value to set too. Leave blank to remove the selected nodes.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        [CakeMethodAlias]
+        public static void XmlPoke(this ICakeContext context, FilePath filePath, string xpath, string value, XmlPokeSettings settings = null)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (filePath == null)
+            {
+                throw new ArgumentNullException("filePath");
+            }
+
+            if (string.IsNullOrWhiteSpace(xpath))
+            {
+                throw new ArgumentNullException("xpath");
+            }
+
+            if (settings == null)
+            {
+                settings = new XmlPokeSettings();
+            }
+
+            var sourceFile = context.FileSystem.GetFile(filePath);
+            if (!sourceFile.Exists)
+            {
+                throw new FileNotFoundException("Source File not found.", sourceFile.Path.FullPath);
+            }
+            
+            using (var sourceReader = sourceFile.Open(FileMode.Open, FileAccess.ReadWrite, FileShare.None))
```

Comment:
Minor as `sourceFile`& `sourceReader` used for both read&write their naming might be a bit misleading, perhaps `file` & `fileStream` is more appropriate.